Sources/SnapshotTesting/AssertSnapshot.swift

```diff
@@ -160,6 +161,7 @@ public func verifySnapshot<Value, Format>(
   record recording: Bool = false,
   snapshotDirectory: String? = nil,
   timeout: TimeInterval = 5,
+  attachmentUserInfo: [AnyHashable: Any]? = nil,
```

Comment:
I wonder if we want to provide some kind of callback instead so that the user info can be derived from the attachment and/or value itself? Not sure attachments are inspectable enough to make this worth it, though.

Reply:
I had considered some sort of mutating callback to modify the attachment but thought the solution was a bit more complicated for not a lot of real extra benefit.

There's not that many parts of the attachment we'd want to otherwise update other than the user info and perhaps (in contrast to the other comment) the lifetime of it's availability.